### PR TITLE
ICU-22158 Make TestAlgorithmicParentFallback() test more robust to di…

### PR DESCRIPTION
**Pull request by richgillam**
…fferent default locales.

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22158
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by markusicu** [comment]
Please retarget this to the https://github.com/unicode-org/icu/tree/maint/maint-72 branch.

File: icu4c/source/test/cintltst/crestst.c [hunk 1]
```diff
@@ -1069,12 +1069,27 @@ static void TestAlgorithmicParentFallback(void) {
         // or sr_Latn to root, but all of these resource bundle files actually exist in the project
     };
     
+    // fallbacks to "root" in the table above actually fall back to the system default locale.
+    // Trying to compare the locale ID from the resource bundle we get back to uloc_getDefault()
+    // has all kinds of problems, so instead we open a resource bundle with a locale ID that we know
+    // will back to the default, save THAT resource bundle's locale ID, and just compare anything
+    // that's supposed to fall back to "root" to THAT.  (If trying to determine the default locale
+    // in this way fails for some reason, we dump out and don't do the rest of the test.)
```
Comment (markusicu) [inline icu4c/source/test/cintltst/crestst.c:1077]:
```suggestion
    // Fallbacks to "root" in the table above actually fall back to the system default locale.
    // Trying to compare the locale ID from the resource bundle we get back to uloc_getDefault()
    // has all kinds of problems, so instead we open a resource bundle with a locale ID that we know
    // will fall back to the default, save THAT resource bundle's locale ID, and just compare anything
    // that's supposed to fall back to "root" to THAT.  (If trying to determine the default locale
    // in this way fails for some reason, we dump out and don't do the rest of the test.)
```

**Review by markusicu — COMMENTED** [review]
This is a good, nice, clean solution! Thanks!

**Comment by richgillam** [comment]
> Please retarget this to the https://github.com/unicode-org/icu/tree/maint/maint-72 branch.

Oops.  Sorry about that.  How do I do that?

**Comment by markusicu** [comment]
> > Please retarget this to the https://github.com/unicode-org/icu/tree/maint/maint-72 branch.
> 
> Oops. Sorry about that. How do I do that? Do I just need to create a new PR, or is there a way to update this one?

You can edit the target branch right in the PR UI, with the edit button next to the PR title. Right now it might not need anything else. You might need to rebase.

**Comment by richgillam** [comment]
>You can edit the target branch right in the PR UI, with the edit button next to the PR title. Right now it might not need anything else. You might need to rebase.

Thank you!  Sorry; I didn't think to try the "Edit" button.  I thought it just edited the title.  My fault.  I've redirected this PR to the main-72 branch, and I've squashed my commits (and fixed a typo in one comment).

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Comment by richgillam** [comment]
Much obliged!  One final question: Is `main` now open for ICU 73 work, and if so, does that mean I have to manually cherry-pick this fix back to `main`?  Or does some copy everything over in bulk at some later point in the process?

**Comment by markusicu** [comment]
> Much obliged! One final question: Is `main` now open for ICU 73 work,

In principle, yes. We don't want disruptive changes for a while, so that cherry-picking between main & branch is not too painful.

> and if so, does that mean I have to manually cherry-pick this fix back to `main`? Or does some copy everything over in bulk at some later point in the process?

The latter. BRS task "Merge maint branch to main"

**Comment by richgillam** [comment]
> > Much obliged! One final question: Is `main` now open for ICU 73 work,
> 
> In principle, yes. We don't want disruptive changes for a while, so that cherry-picking between main & branch is not too painful.

Fair enough.

> > and if so, does that mean I have to manually cherry-pick this fix back to `main`? Or does some copy everything over in bulk at some later point in the process?
> 
> The latter. BRS task "Merge maint branch to main"

Excellent.

Thanks for your patience and help through this process!

**Pull request merged**